packages/integration-karma/test/rendering/slot/index.spec.js

```diff
@@ -0,0 +1,68 @@
+import { createElement } from 'lwc';
+import Container from 'x/container';
+
+describe('slot diffing algorithm', () => {
+    it('should not re-render child component when parent component is re-rendered and api attributes does not change', function() {
+        const elm = createElement('x-container', { is: Container });
+        document.body.appendChild(elm);
+
+        return Promise.resolve()
+            .then(() => {
+                const xChild = elm.shadowRoot.querySelector('x-child');
+
+                expect(xChild.getRenderedTimes()).toBe(1);
+                elm.reRenderElementInShadow();
+                return Promise.resolve();
+            })
+            .then(() => {
+                const xChild = elm.shadowRoot.querySelector('x-child');
+
+                expect(xChild.getRenderedTimes()).toBe(1);
+            });
+    });
+
+    it('should not re-render child component when slot change', function() {
+        const elm = createElement('x-container', { is: Container });
+        document.body.appendChild(elm);
+
+        return Promise.resolve()
+            .then(() => {
+                elm.updateElementInDefaultSlot();
+                elm.updateElementInNamedSlot();
+
+                return Promise.resolve();
+            })
+            .then(() => {
+                const xChild = elm.shadowRoot.querySelector('x-child');
+
+                expect(xChild.getRenderedTimes()).toBe(1);
```

Comment:
this may potentially break some components that today use the `renderedCallback` to know when a slot changes, instead of the `slotchange` event.